rllib/models/torch/torch_distributions.py

```diff
@@ -105,27 +105,26 @@ def __init__(
         assert (probs is None) != (
             logits is None
         ), "Exactly one out of `probs` and `logits` must be set!"
+
+        if logits is not None:
+            assert temperature > 0.0, "Categorical `temperature` must be > 0.0!"
+            _logits = logits / temperature
+            probs = torch.nn.functional.softmax(_logits, dim=-1)
+
         self.probs = probs
         self.logits = logits
+        self.temperature = temperature
         self.one_hot = torch.distributions.one_hot_categorical.OneHotCategorical(
-            probs=self.probs, logits=self.logits
+            probs=probs
         )
-        super().__init__(probs=probs, logits=logits, temperature=temperature)
+        super().__init__(probs=probs)
```

Comment:
Had to make this change so that temperature gets applied to logits **before** we construct both distributions, cat and one hot cat.